### PR TITLE
[Marketo Static Lists] Add audience functions

### PR DESCRIPTION
**Pull request by maryamsharif**
This PR adds the audience functions and authorization for Marketo Static Lists. 

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment
Screenshot 2023-10-30 at 12 53 53 PM
Screenshot 2023-10-30 at 12 54 01 PM
Screenshot 2023-10-30 at 12 54 23 PM

### CODE REVIEW

File: packages/destination-actions/src/destinations/marketo-static-lists/index.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import type { AudienceDestinationDefinition } from '@segment/actions-core'
+import { IntegrationError } from '@segment/actions-core'
 import type { Settings } from './generated-types'
 
 import addToList from './addToList'
 import removeFromList from './removeFromList'
+import { MarketoResonse, getAccessToken } from './constants'
```
Comment (rhall-twilio) [inline packages/destination-actions/src/destinations/marketo-static-lists/index.ts:7]: nit, spelling

File: packages/destination-actions/src/destinations/marketo-static-lists/index.ts [hunk 2]
```diff
@@ -11,15 +13,136 @@ const destination: AudienceDestinationDefinition<Settings> = {
...
-
   audienceFields: {},
-
   audienceConfig: {
     mode: {
       type: 'synced', // Indicates that the audience is synced on some schedule; update as necessary
       full_audience_sync: false // If true, we send the entire audience. If false, we just send the delta.
+    },
+    async createAudience(request, createAudienceInput) {
+      const audience_name = createAudienceInput.audienceName
```
Comment (rhall-twilio) [inline packages/destination-actions/src/destinations/marketo-static-lists/index.ts:60]: style, it's better to be consistent in using either camelCasing or snake_casing.
i think in TypeScript the pattern has generally been camelCasing.

File: packages/destination-actions/src/destinations/marketo-static-lists/index.ts (hunk 1 above)
Comment (marinhero) [reply packages/destination-actions/src/destinations/marketo-static-lists/index.ts:7]: No nit, fix it 😄

File: packages/destination-actions/src/destinations/marketo-static-lists/index.ts [hunk 3]
```diff
@@ -11,15 +13,136 @@ const destination: AudienceDestinationDefinition<Settings> = {
...
+      },
+      folder_name: {
+        label: 'Folder Name',
+        description: 'Name of the folder in which to create static lists.',
+        type: 'string',
+        required: true
+      }
+    },
+    refreshAccessToken: async (request, { settings }) => {
+      return { accessToken: await getAccessToken(request, settings) }
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/marketo-static-lists/index.ts:43]: What happens to this call if it fails? is it retried automatically?

File: packages/destination-actions/src/destinations/marketo-static-lists/constants.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import { RequestClient } from '@segment/actions-core/*'
+import { Settings } from './generated-types'
+
+export interface RefreshTokenResponse {
+  access_token: string
+}
+
+export interface MarketoResonse {
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/marketo-static-lists/constants.ts:8]:
```suggestion
export interface MarketoResponse {
```

**Review by marinhero — CHANGES_REQUESTED** [review]
There's a few things that I would like you to address but overall this looks pretty good! Nice job!

File: packages/destination-actions/src/destinations/marketo-static-lists/index.ts (hunk 3 above)
Comment (maryamsharif) [reply packages/destination-actions/src/destinations/marketo-static-lists/index.ts:49]: The event will not pass and eventually get retried.

**Review by marinhero — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @maryamsharif this PR has been deployed

**Pull request merged**